packages/core/src/browser/widget-open-handler.ts

```diff
@@ -86,16 +86,17 @@ export abstract class WidgetOpenHandler<W extends BaseWidget> implements OpenHan
      * @returns promise of the widget that resolves when the widget has been opened.
      */
     async open(uri: URI, options?: WidgetOpenerOptions): Promise<W> {
-        const widget = await this.getOrCreateWidget(uri, options);
-        await this.doOpen(widget, options);
+        const widgetExists = this.widgetExists(uri, options);
+        const widget = widgetExists ? (await this.getWidget(uri, options))! : await this.createWidget(uri, options);
+        await this.doOpen(widget, widgetExists, options);
         return widget;
     }
-    protected async doOpen(widget: W, options?: WidgetOpenerOptions): Promise<void> {
+    protected async doOpen(widget: W, widgetExists: boolean, options?: WidgetOpenerOptions): Promise<void> {
```

Comment:
You could avoid having to change the signature of this method by adding an optional field to the `WidgetOpenerOptions` interface (or making a new one, something like `WidgetDoOpenerOptions`) that has a `widgetAlreadyCreated` field and adding the `widgetExists` information that way.

Reply:
Thank you for the review.
Do you mean we should not delete `doOpen` override method in `git-commit-detail-open-handler.ts` or we just want to reduce the number of parameters?
I prefer the latter suggestion, i.e. creating `WidgetDoOpenerOptions`, since the `widgetExists` information is not necessary for `open` method. 
However, If we shouldn't delete `doOpen` method, the latter suggestion can't be chosen, since the signature of `doOpen` method in `git-commit-detail-open-handler.ts` and in `widget-open-handler.ts` would be vary.

Reply:
> Thank you for the review.
> Do you mean we should not delete `doOpen` override method in `git-commit-detail-open-handler.ts` or we just want to reduce the number of parameters?

Yes, just reducing the parameters so that this conforms more closely to the signature of `doOpen` in `WidgetOpenHandler` sounds good to me. That means it shouldn't break any downstream functionality - they'll just get the behavior without setting `widgetExists`.